File: app/src/main/java/com/github/goldy1992/mp3player/dagger/modules/service/ContentManagerModule.java
package com.github.goldy1992.mp3player.dagger.modules.service;

import android.content.ContentResolver;
import android.content.Context;
import android.support.v4.media.MediaBrowserCompat.MediaItem;

import com.github.goldy1992.mp3player.commons.MediaItemType;
import com.github.goldy1992.mp3player.service.library.ContentManager;
import com.github.goldy1992.mp3player.service.library.MediaItemTypeIds;

import java.util.List;

import javax.inject.Named;
import javax.inject.Singleton;

import dagger.Module;
import dagger.Provides;

@Module
public class ContentManagerModule {

    @Provides
    @Singleton
    public ContentResolver provideContentResolver(Context context) {
        return context.getContentResolver();
    }

    @Provides
    @Singleton
    @Named("starting_playlist")
    public List<MediaItem> providesInitialPlaylist(ContentManager contentManager, MediaItemTypeIds ids) {
        return contentManager.getPlaylist(ids.getId(MediaItemType.SONGS));
    }


}

File: service/src/main/java/com/github/goldy1992/mp3player/service/dagger/modules/service/ContentManagerModule.kt
package com.github.goldy1992.mp3player.service.dagger.modules.service

import androidx.annotation.OptIn
import androidx.media3.common.util.UnstableApi
import com.github.goldy1992.mp3player.service.library.ContentManager
import com.github.goldy1992.mp3player.service.library.MediaContentManager
import dagger.Binds
import dagger.Module
import dagger.hilt.InstallIn
import dagger.hilt.android.components.ServiceComponent
import dagger.hilt.android.scopes.ServiceScoped

@InstallIn(ServiceComponent::class)
@Module
abstract class ContentManagerModule {

    @OptIn(UnstableApi::class)
    @Binds
    @ServiceScoped
    abstract fun providesContentManager(mediaContentManager: MediaContentManager) : ContentManager

}